sketch_oct9a.ino

```diff
@@ -198,25 +212,49 @@ void readAltIMU() {
   }
 }
 
+char sign(int n) {
+  return n < 0 ? '-' : '+';
+}
+
 void sendDataToRadio() {
-  Frame f = createFrame(
-    last_coords,
-    last_accel,
-    last_gyro,
-    last_alt
+  char buf[79];
+  
+  int pitch = radiansToCappedDegrees(last_gyro.x);
+  int roll = radiansToCappedDegrees(last_gyro.y);
+  int yaw = radiansToCappedDegrees(last_gyro.z);
+
+  int accelXcm = last_accel.x * 100;
+  int accelYcm = last_accel.y * 100;
+  int accelZcm = last_accel.z * 100;
+
+  snprintf(
+    buf,
+    sizeof buf,
+    "%c%08X,%c%08X,%07X,%X,%c%02X,%c%02X,%c%02X,%c%04X,%c%04X,%c%04X,%c%04X"
+#if CAPSULE = 2
+    ",%03X,%c%03X,%02X"
+#endif
+    "\n",
+    sign(last_coords.latitude), abs(last_coords.latitude),
```

Comment:
I understand this isn't a problem, like logic wise it's fine, but why is it done like this? 'sign(thing), abs(thing)'

Reply:
This is a fun thing about C format specifiers.

If I wanted a 3-digit decimal value with `+` or `-` on the beginning, I could just say `%+03d` and pass it a signed integer. However, `%+03X` doesn't do the same thing for hexadecimal. For whatever reason, C can only print positive hexadecimal values. So instead, I have to pass it `%c%03X`, and give either `'+'` or `'-'` followed by the magnitude of the number.

Reply:
huh, that's funny. How did you figure that out?

Reply:
I tried it: https://tio.run/##S9ZNzknMS///XzkzLzmnNCVVwaa4JCUzXy/DjosrM69EITcxM09DU6GaS0GhoAgokKahpKptoGecEpOnpKOga2hkrmmNLheBLFf7/z8A